### PR TITLE
Refactor the Makefile

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:

Refactored the code to run all the generators to avoid any issues for newly added code
- Refactored the `generate` target
- Added comments for the targets
- Removed `hack/verify-starlark.sh` which wasn't used anywhere
- Removed the reference for the tiltfile which isn't used anywhere in the code

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -465,13 +468,6 @@ verify-gen: generate ## Verfiy go generated files are up to date
 		echo "generated files are out of date, run make generate"; exit 1; \
 	fi
 
-.PHONY: verify-manifests
-verify-manifests: manifests ## Verfiy go generated manifests files are up to date
```
Comment (Prajyot-Parab) [inline Makefile:469]: @mkumatag we should preserve this one with `generate-manifests` target right?

Comment (mkumatag) [reply Makefile:469]: This will get tested via `verify-generate` -> `generate` -> `generate-manifests`

**Comment by Prajyot-Parab** [comment]
overall lgtm, awaiting CI failure fixes.

**Comment by mkumatag** [comment]
> overall lgtm, awaiting CI failure fixes.

fixe it, ptal

**Review by Prajyot-Parab — APPROVED** [review]
/lgtm

**Pull request merged**